LMNT/src/ops_misc.c

```diff
@@ -85,7 +85,62 @@ LMNT_ATTR_FAST lmnt_result lmnt_op_assignibv(lmnt_ictx* ctx, lmnt_offset arg1, l
     return LMNT_OK;
 }
 
-LMNT_ATTR_FAST lmnt_result lmnt_op_indexdis(lmnt_ictx* ctx, lmnt_offset arg1, lmnt_offset arg2, lmnt_offset arg3)
+LMNT_ATTR_FAST lmnt_result lmnt_op_dloadiis(lmnt_ictx* ctx, lmnt_offset arg1, lmnt_offset arg2, lmnt_offset arg3)
+{
+    const lmnt_data_section* sec = validated_get_data_section(&ctx->archive, arg1);
+    const lmnt_value* values = validated_get_data_block(&ctx->archive, sec->offset);
+    ctx->stack[arg3] = values[arg2];
+    return LMNT_OK;
+}
+
+LMNT_ATTR_FAST lmnt_result lmnt_op_dloadiiv(lmnt_ictx* ctx, lmnt_offset arg1, lmnt_offset arg2, lmnt_offset arg3)
+{
+    const lmnt_data_section* sec = validated_get_data_section(&ctx->archive, arg1);
+    const lmnt_value* values = validated_get_data_block(&ctx->archive, sec->offset);
+    ctx->stack[arg3 + 0] = values[arg2 + 0];
+    ctx->stack[arg3 + 1] = values[arg2 + 1];
+    ctx->stack[arg3 + 2] = values[arg2 + 2];
+    ctx->stack[arg3 + 3] = values[arg2 + 3];
+    return LMNT_OK;
+}
+
+LMNT_ATTR_FAST lmnt_result lmnt_op_dloadirs(lmnt_ictx* ctx, lmnt_offset arg1, lmnt_offset arg2, lmnt_offset arg3)
+{
+    const lmnt_data_section* sec = validated_get_data_section(&ctx->archive, arg1);
+    const lmnt_value* values = validated_get_data_block(&ctx->archive, sec->offset);
+    size_t arg2v = value_to_size_t(ctx->stack[arg2]);
+    if (arg2v < sec->count) {
+        ctx->stack[arg3] = values[arg2v];
+        return LMNT_OK;
+    } else {
+        return LMNT_ERROR_ACCESS_VIOLATION;
+    }
+}
+
+LMNT_ATTR_FAST lmnt_result lmnt_op_dloadirv(lmnt_ictx* ctx, lmnt_offset arg1, lmnt_offset arg2, lmnt_offset arg3)
+{
+    const lmnt_data_section* sec = validated_get_data_section(&ctx->archive, arg1);
+    const lmnt_value* values = validated_get_data_block(&ctx->archive, sec->offset);
+    size_t arg2v = value_to_size_t(ctx->stack[arg2]);
+    if (sec->count >= 4 && arg2v <= sec->count - 4) {
```

Comment:
I think count here is unsigned (lmnt_offset), could count be less than 4? would that be an issue here, given it would (I think) underflow?